plugin/pkg/scheduler/scheduler.go

```diff
@@ -44,6 +50,33 @@ type Scheduler struct {
 	config *Config
 }
 
+// These are the functions which need to be provided in order to build a Scheduler configuration.
+// An implementation of this can be seen in factory.go.
+type Configurator interface {
```

Comment:
in any case, i think IConfiguration is probably a better name then anything else.  im happy to modify it if you want... +0 since no name is really that good for this strange object that mixes config creation with utility functionality.